### PR TITLE
[DP-6] generate PDF and also some form additions

### PR DESCRIPTION
**Pull request by michael-caktus**
Frontend work. /generate/:pk/ page form inputs are now required. Includes US states and Contacts of type attorney.

Generate button now hits the endpoint and successfully displays PDF in browsers pdf viewer (if it has one... not sure what it does if it doesn't....)

### CODE REVIEW

File: frontend/src/components/pages/GenerationPage/GenerationInputs.js [hunk 1]
```diff
@@ -1,24 +1,40 @@
...
   useEffect(() => {
-    // fetch attornies
-  });
-
-  const mapAttorneysToOptions = () => {
-    return FAKE_ATTORNEYS.map(att => ({ value: att.id, label: att.name }));
-  };
+    (async function() {
+      try {
+        const { data } = await Axios.get('/contact/?category=attorney');
```
Comment (copelco) [inline frontend/src/components/pages/GenerationPage/GenerationInputs.js:31]: Cool!

File: frontend/src/components/pages/GenerationPage/GenerationInputs.js [hunk 2]
```diff
@@ -37,11 +54,27 @@ function GenerationInputs(props) {
...
+          value={licenseNumber}
+          onChange={e => setLicenseNumber(e.target.value)}
+          errors={formErrors?.licenseNumber}
+        />
+      </GenerationInputWrapper>
+
+      <GenerationInputWrapper>
+        <Select
+          label="License state"
+          value={licenseState}
```
Comment (copelco) [inline frontend/src/components/pages/GenerationPage/GenerationInputs.js:73]: Can this default to NC?

File: frontend/src/components/pages/GenerationPage/GeneratePetitionModal/GeneratePetitionModal.js [hunk 3]
```diff
@@ -1,37 +1,84 @@
...
+      // For Firefox it is necessary to delay revoking the ObjectURL
+      window.URL.revokeObjectURL(pdfBlobUrl);
+    }, 100);
+  };
+
+  const handleGenerate = async () => {
+    const derivedPetition = _buildPetition();
+    try {
+      const { data } = await Axios.post('/generate-petition/', derivedPetition, {
+        responseType: 'arraybuffer'
```
Comment (copelco) [inline frontend/src/components/pages/GenerationPage/GeneratePetitionModal/GeneratePetitionModal.js:56]: 🆒

File: frontend/src/components/pages/GenerationPage/GenerationInputs.js (hunk 1 above)
Comment (michael-caktus) [reply frontend/src/components/pages/GenerationPage/GenerationInputs.js:31]: good ol' IIFE

**Pull request merged**